skopt/dummy_opt.py

```diff
@@ -52,20 +67,37 @@ def dummy_minimize(func, dimensions, n_calls=1000, random_state=None):
     """
     rng = check_random_state(random_state)
     space = Space(dimensions)
-    X = space.rvs(n_samples=n_calls, random_state=rng)
-
-    init_y = func(X[0])
+    if x0 is None:
+        x0 = []
+    if type(x0) is not list:
+        x0 = list(x0)
+    if len(x0) > 0 and type(x0[0]) is not list:
+        x0 = [x0]
+    n_random_starts = n_calls
+    if y0 is None:
+        y0 = [func(x) for x in x0]
+        n_random_starts -= len(y0)
+    if type(y0) is not list:
```

Comment:
I think we can support y0 to be an iterable (http://stackoverflow.com/a/1952655/1170730). I'm saying this because you are doing `y = np.array(y)` below.